spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'rails_helper'
+
+RSpec.describe BooksController, type: :controller do
+  describe 'GET /index' do
+    subject { get :index }
+
+    it 'returns all books' do
+      expect(subject.status).to eq(200)
+      expect(assigns(:books)).to eq(Book.all)
+    end
+  end
+
+  describe 'GET /show' do
+    subject { get :show, params: }
+
+    let(:book) { create(:book) }
+    let(:params) { { id: book.id } }
+
+    context 'when book exist' do
+      it 'returns book' do
+        expect(subject.status).to eq(200)
+        expect(subject).to render_template(:show)
+        expect(assigns(:book)).to eq(Book.find(book.id))
+      end
+    end
+
+    context 'when book not exist' do
+      let(:params) { { id: -1 } }
+
+      it 'raise error not found' do
+        # expect(subject.status).to eq(404)
+        expect { subject }.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+  end
+
+  describe 'GET /new' do
+    subject { get :new }
+
+    it 'renders to new book form' do
+      expect(subject.status).to eq(200)
+      expect(subject).to render_template(:new)
+    end
+  end
+
+  describe 'POST /create' do
+    subject { post :create, params: }
+
+    let(:params) { { book: { name: 'Controller book', description: 'Book from controller', release: '2023-01-01' } } }
```

Comment:
```suggestion
    let(:params) do 
      {
        book: { 
          name: 'Controller book', 
          description: 'Book from controller', 
          release: '2023-01-01'
        }
      }
    end
```

We can do this if it too long for 1 line